dbms/src/Columns/ColumnWithDictionary.cpp

```diff
@@ -109,8 +109,8 @@ namespace
 }
 
 
-ColumnWithDictionary::ColumnWithDictionary(MutableColumnPtr && column_unique_, MutableColumnPtr && indexes_)
-    : dictionary(std::move(column_unique_)), idx(std::move(indexes_))
+ColumnWithDictionary::ColumnWithDictionary(MutableColumnPtr && column_unique_, MutableColumnPtr && indexes_, bool is_shared)
```

Comment:
Missing comment about `is_shared`.
We should try to utilize `COWPtr` for this purpose (to unshare `ColumnUnique` on mutating operations).